datafusion/core/src/physical_plan/aggregates/order/mod.rs

```diff
@@ -52,7 +52,6 @@ impl GroupOrdering {
         } = ordering;
 
         Ok(match mode {
-            GroupByOrderMode::None => GroupOrdering::None,
```

Comment:
I haven't reviewed this super carefully yet, but  `GroupOrderingNone` was substantially faster than` GroupOrderingPartial`, because `GroupOrderingPartial` requires additional comparisons per group key.

Thus I would expect this test to slow down performance. I will run some performance tests next week.

Reply:
I should have stressed it, but I recognized that we already store aggregation as option as in [link](https://github.com/apache/arrow-datafusion/blob/01d7dbaf392b4a217e1a27c96bbcd1126b763828/datafusion/core/src/physical_plan/aggregates/mod.rs#L269). When this mode is `None`, it is equivalent to `GroupByOrderMode::None`. So I thought, there is no need to keep track represent this as mode separately. As can be seen in [link](https://github.com/apache/arrow-datafusion/blob/01d7dbaf392b4a217e1a27c96bbcd1126b763828/datafusion/core/src/physical_plan/aggregates/row_hash.rs#L239). When option is `None`, `GroupOrdering::None ` is used. If this is misleading, I can retract this change. It is not important, for this PR.

Reply:
That makes sense. Sorry for my confusion

Reply:
I got inspired / wanted some excuse to work on the code, so I made https://github.com/apache/arrow-datafusion/pull/7150 as a potential alternate approach